src/state/messages/convo.ts

```diff
@@ -66,6 +73,17 @@ export type ConvoState =
       status: ConvoStatus.Destroyed
     }
 
+export function isConvoItemMessage(
+  item: ConvoItem,
+): item is ConvoItem & {type: 'message'} {
+  if (!item) return false
+  return (
+    item.type === 'message' ||
+    item.type === 'deleted-message' ||
+    item.type === 'pending-message'
+  )
+}
+
```

Comment:
I think we want ConvoItem` to always include `sender` so that we can use a single `MessageItem` for rendering these. Then we can handle different logic based on the `type`.